server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -500,6 +501,12 @@ private synchronized IndexService createIndexService(final String reason,
     }
 
     private EngineFactory getEngineFactory(final IndexSettings idxSettings) {
+        final IndexMetaData indexMetaData = idxSettings.getIndexMetaData();
+        if (indexMetaData != null && indexMetaData.getState() == IndexMetaData.State.CLOSE) {
+            // NoOpEngine takes precedence as long as the index is closed
+            return new NoOpEngineFactory();
```

Comment:
this can just be `NoOpEngine::new` then we don't need the factory class altogether?